vyper/types/types.py

```diff
@@ -163,15 +163,18 @@ def __repr__(self):
 
 
 # Data structure for a struct, e.g. {a: <type>, b: <type>}
+# struct can be named or anonymous. name=None indicates anonymous.
 class StructType(NodeType):
-    def __init__(self, members):
+    def __init__(self, members, name=None):
         self.members = copy.copy(members)
+        self.name = name
 
     def eq(self, other):
-        return other.members == self.members
+        return other.name == self.name and other.members == self.members
```

Comment:
I think named and unamed structs should equate to being the same? So I would remove the name check here.

Reply:
I'm kind of leaning towards the name being part of the equality check, especially for type checking. It doesn't really make sense to me for structs to just be aliases. Maybe we should have a separate keyword to distinguish between aliasing and creating a new type, e.g.
```
struct Meters : 
  meters: uint256
# vs
alias Meters = { meters: uint256 }
# vs
newtype Meters = { meters: uint256 } # I think this use case is already covered by `struct Meters`
```

Reply:
Hmm. What is the use case for having an anonymous struct be equal to a named struct? Actually I'm starting to wonder if enforcing that all structs be named would be better for Vyper code readability in general.

Reply:
I think this is more for corner cases, I definitely agree named structs would be much more in common use if available and this concern would be minimal. I know I for one am missing them in my latest draft at implementing Plasma in Vyper.